src/astro/files/locations/base.py

```diff
@@ -0,0 +1,52 @@
+import glob
+import os
+from abc import ABC, abstractmethod
+from typing import List, Optional
+from urllib.parse import urlparse
+
+
+class Location(ABC):
```

Comment:
Should we call this `FileLocation`?

Reply:
It's within the Files folder, so kinda in the right context. But I have no strong opinion on this one.

Reply:
But these will be used in SQL operators, where we _might_ have an API Location in future and we will run into name collision, so `FileLocation` might be better or even `BaseFileLocation` to be consistent with `BaseDatabase` in https://github.com/astro-projects/astro/pull/320/files#diff-5280fa37ed21b32d1f80cf6e5768fe10eccf88b0ea0a33b16a3601ca41f82de1

Reply:
+1 `BaseFileLocation`